### PR TITLE
Refactor some get_assembly and get_study_runs to retry with different portals

### PR DESCRIPTION
**Pull request by mberacochea**
I've updated the pre-commits, removed py3.7 from tests, and added ruff.

### CODE REVIEW

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    branches:
+      - "master"
+      - "develop"
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8, 3.9, '3.10']
```
Comment (arajkovic) [inline .github/workflows/linting.yml:18]: Why are there 3 versions used for linting?

File: ena_portal_api/ena_handler.py [hunk 2]
```diff
@@ -406,13 +401,16 @@ def get_run(
...
+            or response.status_code == requests.codes.ok
+            and not response.json()
+        ):
             if attempt < 2:
                 attempt += 1
                 sleep(1)
                 logging.warning(
-                    "Error 204 when retrieving run {} in dataPortal {}, "
-                    "retrying {}".format(run_accession, data.get("dataPortal"), attempt)
+                    f"Error 204 when retrieving run {run_accession} in dataPortal {data.get('dataPortal')}, retrying {attempt}"
```
Comment (arajkovic) [inline ena_portal_api/ena_handler.py:413]: :perfection:

**Review by KateSakharova — APPROVED** [review]
makes sense

File: ena_portal_api/ena_handler.py (hunk 2 above)
Comment (mberacochea) [reply ena_portal_api/ena_handler.py:413]: Sort of. This is to address some data not being available in one data portal (ENA has several) but being available in another. Also, sometimes the response is 404, or 204... or even 200 with an empty (several different cases depending in the kind of data)

**Pull request merged**